rclpy/src/rclpy/_rclpy.c

```diff
@@ -1671,16 +1671,28 @@ rclpy_time_since_last_call(PyObject * Py_UNUSED(self), PyObject * args)
 static void
 _rclpy_destroy_subscription(PyObject * pyentity)
 {
-  if (PyCapsule_IsValid(pyentity, "rclpy_subscription_t")) {
-    rclpy_subscription_t * sub = (rclpy_subscription_t *)PyCapsule_GetPointer(
-      pyentity, "rclpy_subscription_t");
-    if (!sub) {
-      return;
-    }
-    rcl_ret_t ret = rcl_subscription_fini(&(sub->subscription), sub->node);
-    (void)ret;
-    PyMem_Free(sub);
+  if (!PyCapsule_IsValid(pyentity, "rclpy_subscription_t")) {
+    // Warning should use line number of the current stack frame
+    int stack_level = 1;
+    PyErr_WarnFormat(
+      PyExc_RuntimeWarning, stack_level, "_rclpy_destroy_subscrition failed to get pointer");
+    return;
+  }
+
+  rclpy_subscription_t * sub = (rclpy_subscription_t *)PyCapsule_GetPointer(
+    pyentity, "rclpy_subscription_t");
+  // ...GetPointer() is guaranteed to return non NULL if PyCapsule_IsValid passes
```

Comment:
How about just calling `PyCapsule_GetPointer` and if it returns `NULL` reset the error, print the warning and return early?

Reply:
e0c9c21